### PR TITLE
OnGoingMatches should be stored in memory instead of MongoDB

### PR DESCRIPTION
**Pull request by mourasman**

### CODE REVIEW

File: W3ChampionsStatisticService/Matches/MatchRepository.cs [hunk 1]
```diff
@@ -162,60 +165,63 @@ private PlayerScore CreateDetail(PlayerBlizzard playerBlizzard)
...
         {
-            return CreateCollection<OnGoingMatchup>()
-                .CountDocumentsAsync(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
-                && (gateWay == GateWay.Undefined || m.GateWay == gateWay));
+            return _onGoingMatchups.FindAll(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
+                                                 && (gateWay == GateWay.Undefined || m.GateWay == gateWay)).Count;
         }
     }
+
+    internal static class OnGoingMatchUps
```
Comment (abeninski) [inline W3ChampionsStatisticService/Matches/MatchRepository.cs:219]: maybe rename this to OnGoingMatchesMemoryStore?

File: W3ChampionsStatisticService/Matches/MatchRepository.cs [hunk 2]
```diff
@@ -162,60 +165,63 @@ private PlayerScore CreateDetail(PlayerBlizzard playerBlizzard)
...
+    {
+        static OnGoingMatchUps()
+        {
+        }
+
+        public static List<OnGoingMatchup> Instance { get; } = new List<OnGoingMatchup>();
```
Comment (abeninski) [inline W3ChampionsStatisticService/Matches/MatchRepository.cs:225]: and rename this to `Matches`?

File: W3ChampionsStatisticService/Matches/MatchRepository.cs [hunk 3]
```diff
@@ -13,8 +13,11 @@ namespace W3ChampionsStatisticService.Matches
 {
     public class MatchRepository : MongoDbRepositoryBase, IMatchRepository
     {
+        private readonly List<OnGoingMatchup> _onGoingMatchups;
```
Comment (abeninski) [inline W3ChampionsStatisticService/Matches/MatchRepository.cs:16]: You don't need to assign this? Just use OnGoingMatchesMemoryStore.Matches

**Review by abeninski — CHANGES_REQUESTED** [review]

File: W3ChampionsStatisticService/Matches/MatchRepository.cs (hunk 1 above)
Comment (mourasman) [reply W3ChampionsStatisticService/Matches/MatchRepository.cs:219]: yep, sure

File: W3ChampionsStatisticService/Matches/MatchRepository.cs (hunk 3 above)
Comment (mourasman) [reply W3ChampionsStatisticService/Matches/MatchRepository.cs]: fixed

File: W3ChampionsStatisticService/Matches/MatchRepository.cs (hunk 2 above)
Comment (mourasman) [reply W3ChampionsStatisticService/Matches/MatchRepository.cs]: done

**Review by abeninski — APPROVED** [review]

**Pull request merged**